### PR TITLE
remove references to unique ptrs

### PR DESCRIPTION
**Pull request by jacquelinekay**
Connects to ros2/rclcpp#119

### CODE REVIEW

**Comment by esteve** [comment]
+1

**Comment by dirk-thomas** [comment]
+1

File: intra_process_demo/src/cyclic_pipeline/cyclic_pipeline.cpp [hunk 1]
```diff
@@ -29,7 +29,7 @@ struct IncrementerPipe : public rclcpp::Node
     // Create a subscription on the input topic.
     sub = this->create_subscription_with_unique_ptr_callback<std_msgs::msg::Int32>(
       in, rmw_qos_profile_default,
-      [this](std_msgs::msg::Int32::UniquePtr & msg) {
+      [this](std_msgs::msg::Int32::UniquePtr msg) {
```
Comment (esteve) [inline intra_process_demo/src/cyclic_pipeline/cyclic_pipeline.cpp]: I know this wasn't introduced by this PR, but couldn't this callback cause an invalid pointer by passing `this` when `IncrementerPipe` goes out of scope?

Comment (esteve) [inline intra_process_demo/src/cyclic_pipeline/cyclic_pipeline.cpp]: I've made the changes to not pass the `this` pointer in https://github.com/ros2/demos/tree/avoid-this-lambda you can merge it or if you'd rather not, I can create a separate PR for it.

Comment (jacquelinekay) [inline intra_process_demo/src/cyclic_pipeline/cyclic_pipeline.cpp]: I'll put it off for another PR. Thanks for fixing it.

Comment (esteve) [inline intra_process_demo/src/cyclic_pipeline/cyclic_pipeline.cpp]: No problem, I'll create a separate PR for that branch.

Comment (esteve) [inline intra_process_demo/src/cyclic_pipeline/cyclic_pipeline.cpp]: Done: https://github.com/ros2/demos/pull/28

**Pull request merged**